blocks_vertical/extensions.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2016 Massachusetts Institute of Technology
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+goog.provide('Blockly.Blocks.extensions');
+
+goog.require('Blockly.Blocks');
+goog.require('Blockly.Colours');
+goog.require('Blockly.constants');
+goog.require('Blockly.ScratchBlocks.VerticalExtensions');
+
+
+Blockly.Blocks['extension_pen_down'] = {
+  /**
+   * @this Blockly.Block
+   */
+  init: function() {
```

Comment:
You can use Blockly.defineBlocksWithJsonArray for this: https://github.com/LLK/scratch-blocks/blob/develop/core/blockly.js#L358

Reply:
I just followed the pattern of the rest of the blocks definition files in that directory. It looks like defineBlocksWithJsonArray is used mainly in the unit tests. Is it okay if I leave it as is? Or is there some reason that defineBlocksWithJsonArray is better?

Reply:
Discussed offline; doesn't need to happen in this PR, but should happen eventually.